src/features/transaction-list.tsx

```diff
@@ -27,8 +28,13 @@ export const TransactionList = (props: TransactionListProps) => {
         : undefined,
     [data]
   );
+  const router = useRouter();
+  const filterScope =
```

Comment:
We could maybe extract this logic as it's the same as the one above and we might extend this later to other pages.